components/admin_console/schema_admin_settings.jsx

```diff
@@ -787,7 +797,10 @@ export default class SchemaAdminSettings extends React.Component {
         this.setState({errorTooltip: isElipsis});
     }
 
-    doSubmit = (callback, getStateFromConfig) => {
+    doSubmit = async (callback, getStateFromConfig) => {
```

Comment:
The changes here are converting this "generic" approach into a specific case handling approach. I'm not against the idea, but I'm against make it specific to the BrandImage. By the way, you are reusing the same reference or multiple custom element which is going to break if you have other custom element in the page after this one.

So, you have 2 problems here, 1 reference that is used for multiple components (that shouldn't happen), and one handling everything specifically for the "CustomBrandImage" component, which shouldn't be.

My proposal, if we want to follow this path, is to have an array of references, and for each reference try to run the an handleSave function.

something like:
```
const customComponent = this.customComponentRef.current;
if (customComponent.handleSave) {
    result = await customComponent.handleSave()
}
```

Other important thing, you are accessing to the "local state" of the component from the outside, this is allowed by the references, but because you can don't mean you should. The local state should be maintained private, so I would restrict the usage of the references to call the concrete methods of the components.

Reply:
@jespino Thank you so much for pointing these 2 problems out. I hadn't thought of referencing multiple components being an issue but I did have the issue of making a generic function specific to the `CustomBrandImage` component. I've addressed both of these problems by making it more general and having an array of references. 

I'll also be sure to keep in mind not to access state from outside the component too!
Thanks for your feedback and help 😃

Reply:
I don't know if we can keep track of multiple refs by pushing them onto an array. When using a ref callback like this, it gets called with `null` when the child component is unmounted. It's easy enough to avoid adding a null ref to that array, but I'm wondering if we might run into any problems because we have no way to remove entries from that array.

I'd also be worried about using refs here because they can get a bit tricky with higher order components like ones created using `connect(MyComponent)` to connect to the redux state, but that's just a problem with refs in general.

@jespino What's the motivation for preferring refs over something like I suggested where the `SchemaAdminSettings` passes a method into the child component that the child can use to tell the `SchemaAdminSettings` that it has something to save? I'm 0/5 either way, but that might be easier than using refs

Reply:
The problem with that approach is how hard to follow is it (passing a function that allow to set the function used to save from the parent), but yes, the "array of refs" looks like complicated and more error prone. I'm 2/5 here, but I guess your solution, for this case is not perfect, but is less error prone so I think I prefer it.

Reply:
Requested changes have been made 🎉